src/sentry/static/sentry/app/components/events/interfaces/spans/filter.tsx

```diff
@@ -0,0 +1,342 @@
+
+    return result;
+  }
+
+  isOperationNameActive(operationName: string): boolean {
+    const {operationNameFilters} = this.props;
+
+    if (operationNameFilters.type === 'no_filter') {
+      return false;
+    }
+
+    // invariant: operationNameFilters.type === 'active_filter'
+
+    return operationNameFilters.operationNames.has(operationName);
+  }
+
+  getNumberOfActiveFilters(): number {
+    const {operationNameFilters} = this.props;
+
+    if (operationNameFilters.type === 'no_filter') {
+      return 0;
+    }
+
+    return operationNameFilters.operationNames.size;
+  }
+
+  render() {
+    const operationNames = this.getOperationNames();
+
+    if (operationNames.length === 0) {
+      return null;
+    }
+
+    const checkedQuantity = this.getNumberOfActiveFilters();
+
+    const buttonProps = {
+      label: (
+        <React.Fragment>
+          <IconFilter size="xs" />
+          <FilterLabel>{t('Filter')}</FilterLabel>
+        </React.Fragment>
+      ),
+      priority: 'default',
+      hasDarkBorderBottomColor: false,
+    };
+
+    if (checkedQuantity > 0) {
+      buttonProps.label = (
+        <span>{tn('%s Active Filter', '%s Active Filters', checkedQuantity)}</span>
+      );
+      buttonProps.priority = 'primary';
+      buttonProps.hasDarkBorderBottomColor = true;
+    }
+
+    return (
+      <Wrapper>
+        <DropdownControl
+          menuWidth="240px"
+          blendWithActor
+          button={({isOpen, getActorProps}) => (
+            <StyledDropdownButton
+              {...getActorProps()}
+              showChevron={false}
+              isOpen={isOpen}
+              hasDarkBorderBottomColor={buttonProps.hasDarkBorderBottomColor}
+              priority={buttonProps.priority as DropdownButtonProps['priority']}
+            >
+              {buttonProps.label}
+            </StyledDropdownButton>
+          )}
+        >
+          <MenuContent
+            onClick={event => {
+              // propagated clicks will dismiss the menu; we stop this here
+              event.stopPropagation();
+            }}
+          >
+            <Header>
+              <span>{t('Operation')}</span>
+              <CheckboxFancy
+                isChecked={checkedQuantity > 0}
+                isIndeterminate={
+                  checkedQuantity > 0 && checkedQuantity !== operationNames.length
+                }
+                onClick={event => {
+                  event.stopPropagation();
+                  this.props.toggleAllOperationNameFilters(operationNames);
+                }}
+              />
+            </Header>
+            <List>
+              {operationNames.map((operationName: string) => {
+                const isActive = this.isOperationNameActive(operationName);
+
+                return (
+                  <ListItem key={operationName} isChecked={isActive}>
+                    <OperationDot
+                      style={{backgroundColor: pickSpanBarColour(operationName)}}
```

Comment:
this is a styled component, maybe you would prefer to pass only the `backgroundColor` as a prop?